File: src/main/java/com/googongill/aditory/controller/dto/category/response/CopyCategoryResponse.java
package com.googongill.aditory.controller.dto.category.response;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.googongill.aditory.domain.enums.CategoryState;
import com.googongill.aditory.service.dto.category.CopyCategoryResult;
import lombok.Builder;

import java.time.LocalDateTime;

@Builder
@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class CopyCategoryResponse{
    private Long categoryId;
    private String categoryName;
    private String asCategoryName;
    private CategoryState categoryState;
    private LocalDateTime createdAt;

    public static CopyCategoryResponse of(CopyCategoryResult copyCategoryResult) {
        return CopyCategoryResponse.builder()
                .categoryId(copyCategoryResult.getCategoryId())
                .categoryName(copyCategoryResult.getCategoryName())
                .asCategoryName(copyCategoryResult.getAsCategoryName())
                .categoryState(copyCategoryResult.getCategoryState())
                .createdAt(copyCategoryResult.getCreatedAt())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/controller/dto/category/response/CopyCategoryResponse.kt
package com.googongill.aditory.controller.dto.category.response

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.googongill.aditory.domain.enums.CategoryState
import com.googongill.aditory.service.dto.category.CopyCategoryResult
import java.time.LocalDateTime

@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
data class CopyCategoryResponse(
    val categoryId: Long?,
    val categoryName: String,
    val asCategoryName: String,
    val categoryState: CategoryState?,
    val createdAt: LocalDateTime?
) {

    companion object {
        fun of(copyCategoryResult: CopyCategoryResult): CopyCategoryResponse {
            return CopyCategoryResponse(
                categoryId = copyCategoryResult.categoryId,
                categoryName = copyCategoryResult.categoryName,
                asCategoryName = copyCategoryResult.asCategoryName,
                categoryState = copyCategoryResult.categoryState,
                createdAt = copyCategoryResult.createdAt
            )
        }
    }

}
